### PR TITLE
sends an email to viewer to set password when a viewer account is created

### PR DESCRIPTION
**Pull request by bastienk**

### CODE REVIEW

**Review by zanchi-r — CHANGES_REQUESTED** [review]
I was not able to compile this:
src/github.com/trackit/trackit-server/users/create.go:147:11: undefined: uuid
src/github.com/trackit/trackit-server/users/create.go:150:3: undefined: logger
src/github.com/trackit/trackit-server/users/create.go:169:12: undefined: time
src/github.com/trackit/trackit-server/users/create.go:173:3: undefined: logger
src/github.com/trackit/trackit-server/users/create.go:180:3: undefined: logger

File: users/create.go [hunk 1]
```diff
@@ -156,6 +167,23 @@ func createViewerUser(request *http.Request, a routes.Arguments) (int, interface
...
+		Token:   tokenHash,
+		Created: time.Now(),
+	}
+	err = dbForgottenPassword.Insert(tx)
+	if err != nil {
+		logger.Error("Failed to insert viewer password token in database.", err.Error())
+		return 500, errors.New("Failed to create viewer password token")
+	}
+	mailSubject := "Your TrackIt viewer password"
+	mailBody := fmt.Sprintf("Please follow this link to create your password: https://re.trackit.io/reset/%d/%s.", viewerUser.Id, token)
```
Comment (zanchi-r) [inline users/create.go]: The first parameter should be dbForgottenPassword.ID instead of viewerUser.Id

**Review by zanchi-r — CHANGES_REQUESTED** [review]
Just a small change and the PR will be ready

**Comment by bastienk** [comment]
I pushed the changes you required

**Pull request merged**